torch/csrc/autograd/variable.cpp

```diff
@@ -170,7 +170,7 @@ void Variable::Impl::set_data(const at::Tensor &new_data) {
   device_opt_ = new_data.device();
   type_id_ = new_data.dispatch_type().type_id();
 
-  auto new_data_impl_copy = new_data.getIntrusivePtr()->shallow_copy_and_detach();
+  auto new_data_impl_copy = new_data.getIntrusivePtr()->shallow_copy_and_detach(/*version_counter=*/0);
```

Comment:
I don't understand -- don't you want to pass in exactly what `saved_version_counter` is below?

Reply:
Sorry it was an oversight -- fixed.